### PR TITLE
Using wait_for_expected_group_state instead of sleeping

### LINKED ISSUE #1404: Make cron scheduler tests deterministic
This can be done by using `wait_for_expected_group_state` with short interval instead of sleeping for `60 + scheduler_interval` seconds and then calling `verify_group_state`. This is already done in few tests like `test_system_cron_style_policy_executes_again`.

### PR DESCRIPTION
**Pull request by manishtomar**
Closes #1404. The tests now wait for expected group state for maximum scheduled time. This way the tests will wake up at beginning of minute in cron based tests and continue accordingly.

### CODE REVIEW

File: autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_execute_scheduler_negative.py [hunk 1]
```diff
@@ -152,7 +154,9 @@ def test_system_scheduler_batch(self):
             sp_list=at_style_policies_list)
         group = response.entity
         self.resources.add(group, self.empty_scaling_group)
-        sleep(self.scheduler_interval * 2)
+        # An extra 1 second to let scheduler processing take place due to
+        # excess events
+        sleep(10 + self.scheduler_interval + 1)
```
Comment (cyli) [inline autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_execute_scheduler_negative.py]: Non-blocking question:  why the change from twice the scheduler interval to 10 + one interval + 1?  Is the scheduler interval just 10?

File: autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_scheduler_update_group.py [hunk 2]
```diff
@@ -143,23 +145,23 @@ def test_system_upd_launch_config_cron_style_scheduler(self):
...
             schedule_cron='* * * * *')
-        sleep(60 + self.scheduler_interval)
         active_servers = self.sp_change + group.groupConfiguration.minEntities
         active_after_scaleup = self.wait_for_expected_number_of_active_servers(
-            group_id=group.id,
-            expected_servers=active_servers)
-        upd_lc_server = set(
-            active_after_scaleup) - set(active_list_b4_upd)
+            group_id=group.id, expected_servers=active_servers,
+            timeout=(60 + self.scheduler_interval), interval_time=2,
```
Comment (cyli) [inline autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_scheduler_update_group.py]: Is this enough time?  The original sleeps for 60+scheduler, and then waits for the number to be reached.  The new version just waits 60+scheduler seconds.  Should a few extra seconds be added for a safety margin?

**Comment by cyli** [comment]
I seem to be having trouble running the `test_system_multi_scheduler_webhook_policies` tests

**Comment by manishtomar** [comment]
@cyli Yeah. That test broke during linting. I am fixing it now.

File: autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_scheduler_update_group.py (hunk 2 above)
Comment (manishtomar) [inline autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_scheduler_update_group.py]: Yeah. You are right. There should be a few seconds margin for safety. I'll change it in all the tests. Thanks :)

**Comment by cyli** [comment]
This otherwise LGTM, since the test fix is in the other PR.  :)

File: autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_execute_scheduler_negative.py (hunk 1 above)
Comment (manishtomar) [inline autoscale_cloudroast/test_repo/autoscale/system/schedule_policies/test_system_execute_scheduler_negative.py]: No. The 10 account for the 10s in future when the policy gets executed.

**Comment by manishtomar** [comment]
I'll run all scheduler tests and remove "in progress" after that.

**Comment by manishtomar** [comment]
All the tests pass in dev vm

**Comment by cyli** [comment]
@manishtomar cool, all the tests passed for me previously too except for `test_system_multi_scheduler_webhook_policies`, which has now been fixed.

LGTM!  Thank you for making our tests more dependable!

**Comment by manishtomar** [comment]
> Thank you for making our tests more dependable!

@cyli We'll see about that :)
